PythonPonies/bangazon-storefront
# Initial commit for user/customer creation, login, logout functions

# Description
This PR includes features to create user/customer objects on a registration page, login, and logout. 

## Number of Fixes
N/A

## Related Ticket(s)
N/A

## Problem to Solve
Our storefront requires persistent user data, with users having the ability to securely log in and create personalized orders with payment information.

## Proposed Changes
Describe the proposed changes.

## Expected Behavior
The register form stores customer input as data in the DB and redirects the user to the productTypes template. Login allows returning users to access their account info, and also redirects to the productTypes template. If the user attempts to login but does not exist in the DB, they are returned to the index page. 

## Steps to Test Solution

1. Drop your DB
2. (In the terminal) python manage.py makemigrations
3. python manage.py migrate
4. python manage.py test bangazon_storefront.tests.test_customer_view
5. python manage.py runserver
6. (in the browser) localhost: 8000/register
7.  localhost: 8000/login 
8. Check DB to ensure that customer/user info is stored 

## Testing

[x] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[x] I certify that all existing tests pass

## Documentation
[x] There is new documentation in this pull request that must be reviewed..
[x] I added documentation for any new classes/methods

## Deploy Notes
N/A
